src/wallet/walletutil.cpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "wallet/walletutil.h"
+
+fs::path GetWalletDir()
+{
+    fs::path path;
+
+    if (gArgs.IsArgSet("-walletdir")) {
+        path = fs::system_complete(gArgs.GetArg("-walletdir", ""));
+        if (!fs::is_directory(path)) {
+            // If the path specified doesn't exist, we return the deliberately
+            // invalid empty string.
```

Comment:
Why? Couldn't create it? Is it bad setting?

Reply:
Unlike the datadir, `walletdir` is not created if it doesn't exist. This throws an error in init:
```
$ src/bitcoind -regtest -walletdir=/tmp/walletsdf
Error: Error: Specified wallet directory "/tmp/walletsdf" does not exist.
```
That's the right behavior, IMO.

Reply:
Nit, if `/tmp/walletsdf` is non-directory then the error could be different.